added return docstring for search

iamrajiv opened the pull request:
<!-- This comments are hidden when you submit the pull request, so you do not need to remove them!
Please be sure to check out our contributing guidelines, https://github.com/sunpy/sunpy/blob/master/CONTRIBUTING.rst.
Please be sure to check out our code of conduct, https://github.com/sunpy/sunpy/blob/master/CODE_OF_CONDUCT.rst. -->

<!-- Please just have a quick search on GitHub to see if a similar pull request has already been posted.
We have old closed pull requests that might provide useful code or ideas that directly tie in with your pull request. -->

<!-- We have several automatic features that run when a pull request is open.
They can appear daunting but do not worry about them!
We have a brief explanation of them in the documentation, https://docs.sunpy.org/en/latest/dev_guide/pr_review_procedure.html#continuous-integration. -->

### Description
<!-- Provide a general description of what your pull request does. -->
Fixes the documentation of `search()` by adding `Returns` docstring.
<!-- If the pull request closes any open issues you can add this.
If you replace <Issue Number> with a number GitHub will automatically link it.
If it doesn't, please remove the following line. -->

Fixes #3546

dstansby commented on sunpy/database/database.py:
```diff
@@ -578,6 +578,11 @@ def search(self, *query, **kwargs):
 
+        Returns
+        -------
+        table
```
```suggestion
        table : list
```

dstansby commented on sunpy/database/database.py:
```diff
@@ -578,6 +578,11 @@ def search(self, *query, **kwargs):
+        Returns
+        -------
+        table
+            List of database entries that satisfy all of the given attributes.
```
```suggestion
            List of :class:`sunpy.database.tables.DatabaseEntry` that satisfy all of the given attributes.
```

iamrajiv commented:
@dstansby I have updated the code

iamrajiv commented:
@dstansby I have updated the code please take a look

Cadair commented on changelog/3546.doc.rst:
```diff
@@ -0,0 +1 @@
+Fixed documentation of `search()` by adding `Returns` docstring.
```
```suggestion
Fixed documentation of `sunpy.database.Database.search` by adding ``Returns`` docstring.
```

Cadair commented on sunpy/database/database.py:
```diff
@@ -578,6 +578,12 @@ def search(self, *query, **kwargs):
+        Returns
+        -------
+        table : `list`
+            List of :class:`sunpy.database.tables.DatabaseEntry` that
```
```suggestion
            List of `sunpy.database.tables.DatabaseEntry` that
```

Cadair commented:
hey @iamrajiv I have added a couple of little syntax tweaks. Also the number in the changelog file should refer to the PR and not the issue :smile:

iamrajiv commented:
> hey @iamrajiv I have added a couple of little syntax tweaks. Also the number in the changelog file should refer to the PR and not the issue

sorry for inconvenience @Cadair and thank you for guidance. I have implemented your suggestion please take a look
